core/src/test/scala/unit/kafka/server/ApiVersionsRequestTest.scala

```diff
@@ -17,32 +17,72 @@
+    serverProperties
+  }
+
+  def testApiVersionsRequestTemplate(clusterGenerator: ClusterGenerator): Unit = {
+    val serverProperties: util.HashMap[String, String] = controlPlaneListenerProperties()
+    serverProperties.put("unstable.api.versions.enable", "false")
+    serverProperties.put("unstable.metadata.versions.enable", "true")
+    clusterGenerator.accept(ClusterConfig.defaultBuilder()
+      .setServerProperties(serverProperties)
+      .setMetadataVersion(MetadataVersion.IBP_3_8_IV0)
+      .build())
+  }
+
+  def testApiVersionsRequestIncludesUnreleasedApisTemplate(clusterGenerator: ClusterGenerator): Unit = {
+    val serverProperties: util.HashMap[String, String] = controlPlaneListenerProperties()
+    serverProperties.put("unstable.api.versions.enable", "true")
+    serverProperties.put("unstable.metadata.versions.enable", "true")
+    clusterGenerator.accept(ClusterConfig.defaultBuilder()
+      .setServerProperties(serverProperties)
+      .build())
+  }
+
+  def testApiVersionsRequestValidationV0Template(clusterGenerator: ClusterGenerator): Unit = {
+    val serverProperties: util.HashMap[String, String] = controlPlaneListenerProperties()
+    serverProperties.put("unstable.api.versions.enable", "false")
+    serverProperties.put("unstable.metadata.versions.enable", "false")
+
+    clusterGenerator.accept(ClusterConfig.defaultBuilder()
+      .setServerProperties(serverProperties)
+      .setMetadataVersion(MetadataVersion.IBP_3_7_IV4)
+      .build())
+  }
+
+  def zkApiVersionsRequest(clusterGenerator: ClusterGenerator): Unit = {
+    clusterGenerator.accept(ClusterConfig.defaultBuilder()
+      .setServerProperties(controlPlaneListenerProperties())
+      .build())
+  }
+}
+
 @ExtendWith(value = Array(classOf[ClusterTestExtensions]))
-@ClusterTestDefaults(brokers = 1)
+@ClusterTestDefaults()
 class ApiVersionsRequestTest(cluster: ClusterInstance) extends AbstractApiVersionsRequestTest(cluster) {
 
-  @ClusterTests(Array(
-    new ClusterTest(clusterType = Type.ZK, metadataVersion = MetadataVersion.IBP_3_8_IV0, serverProperties = Array(
-      new ClusterConfigProperty(key = "unstable.api.versions.enable", value = "false"),
-      new ClusterConfigProperty(key = "unstable.metadata.versions.enable", value = "true"),
-      // Configure control plane listener to make sure we have separate listeners for testing.
-      new ClusterConfigProperty(key = "control.plane.listener.name", value = "CONTROL_PLANE"),
-      new ClusterConfigProperty(key = "listener.security.protocol.map", value = "CONTROL_PLANE:PLAINTEXT,PLAINTEXT:PLAINTEXT"),
-      new ClusterConfigProperty(key = "listeners", value = "PLAINTEXT://localhost:0,CONTROL_PLANE://localhost:0"),
-      new ClusterConfigProperty(key = "advertised.listeners", value = "PLAINTEXT://localhost:0,CONTROL_PLANE://localhost:0"),
-    )),
+  @ClusterTests(clusterTemplate = new ClusterTemplate("testApiVersionsRequestTemplate"), value = Array(
```

Comment:
After #15897, we can rewrite this by following style.
```scala
  @ClusterTemplate("testApiVersionsRequestIncludesUnreleasedApisTemplate")
  @ClusterTest(clusterTypes = {Type.CO_KRAFT, Type.KRAFT}, serverProperties = Array(
    new ClusterConfigProperty(key = "unstable.api.versions.enable", value = "false"),
    new ClusterConfigProperty(key = "unstable.metadata.versions.enable", value = "true"),
  ))
```
Maybe we don't need to enhance `ClusterTests` if we allow users to define both ClusterTest and ClusterTemplate. WDYT?

Reply:
I'm fine with this. Will close the pr and JIRA ticket and filed a new minor pull-request to simplify test cases in  ApiVersionsRequestTest.scala